src/arguments.hpp

```diff
@@ -0,0 +1,189 @@
+     */
+    const char* asOneOf(const std::initializer_list<const char*>& expected);
+
+    /**
+     * @brief Get current argument as unsigned number.
+     *        Argument pointer will be moved to the next entry.
+     *
+     * @throw std::invalid_argument if there are no more arguments to handle
+     *                              or argument is not a numeric
+     *
+     * @return argument value as unsigned number
+     */
+    size_t asDigit();
+
+    /**
+     * @brief Get current argument as action.
+     *        Argument pointer will be moved to the next entry.
+     *
+     * @throw std::invalid_argument if there are no more arguments to handle
+     *                              or argument is not an action
+     *
+     * @return argument value as action type
+     */
+    Action asAction();
+
+    /**
+     * @brief Get current argument as toggle.
+     *        Argument pointer will be moved to the next entry.
+     *
+     * @throw std::invalid_argument if there are no more arguments to handle
+     *                              or argument is not an toggle
+     *
+     * @return argument value as toggle state
+     */
+    Toggle asToggle();
+
+    /**
+     * @brief Get current argument as MAC address.
+     *        Argument pointer will be moved to the next entry.
+     *
+     * @throw std::invalid_argument if there are no more arguments to handle
+     *                              or argument is not a valid MAC address
+     *
+     * @return argument value: MAC address
+     */
+    const char* asMacAddress();
+
+    /**
+     * @brief Get current argument as IP address.
+     *        Argument pointer will be moved to the next entry.
+     *
+     * @throw std::invalid_argument if there are no more arguments to handle
+     *                              or argument is not a valid IP address
+     *
+     * @return argument value: IP version and address
+     */
+    std::tuple<IpVer, const char*> asIpAddress();
+
+    /**
+     * @brief Get current argument as IP address with mask.
+     *        Argument pointer will be moved to the next entry.
+     *
+     * @throw std::invalid_argument if there are no more arguments to handle
+     *                              or argument has invalid format
+     *
+     * @return argument value: IP version, address and mask as bits count
+     */
+    std::tuple<IpVer, std::string, uint8_t> asIpAddrMask();
+
+    /**
+     * @brief Check for numeric format.
+     *
+     * @param[in] arg value to check
+     *
+     * @return false if checked value is not a number
+     */
+    static bool isDigit(const char* arg);
+
+    /**
+     * @brief Check for IP address.
+     *
+     * @param[in] arg value to check
+     *
+     * @return IP version or nullopt if checked value is not a valid IP address
+     */
+    static std::optional<IpVer> isIpAddress(const char* arg);
+
+  private:
+    /** @brief Index of the next argument. */
+    size_t index;
```

Comment:
`index` is used in cpp. I don't think we need an iterator here, vector is an indexed array.

Reply:
index is used, but it's numeric value isn't. If it was an iterator, nothing would change for the code using it, but the mechanics behind it would be more obscure, just like C++ likes it to be. :)